src/topology/topological_fiber_bundle.lean

```diff
@@ -105,135 +105,235 @@ Fiber bundle, topological bundle, vector bundle, local trivialization, structure
-    (inter_subset_right _ _),
-  exact is_open_map_fst _ this
+  refine is_open_map_iff_nhds_le.2 (λ x, _),
+  rcases h (proj x) with ⟨e, ex⟩,
+  refine (e.map_proj_nhds _).ge,
+  rwa e.source_eq
 end
 
+/-- The first projection in a product is a trivial topological fiber bundle. -/
+lemma is_trivial_topological_fiber_bundle_fst :
+  is_trivial_topological_fiber_bundle F (prod.fst : B × F → B) :=
+⟨homeomorph.refl _, λ x, rfl⟩
+
 /-- The first projection in a product is a topological fiber bundle. -/
 lemma is_topological_fiber_bundle_fst : is_topological_fiber_bundle F (prod.fst : B × F → B) :=
-begin
-  let F : bundle_trivialization F (prod.fst : B × F → B) :=
-  { base_set      := univ,
-    open_base_set := is_open_univ,
-    source_eq     := rfl,
-    target_eq     := by simp,
-    proj_to_fun   := by simp,
-     ..local_homeomorph.refl _ },
-  exact λx, ⟨F, by simp⟩
-end
+is_trivial_topological_fiber_bundle_fst.is_topological_fiber_bundle
+
+/-- The second projection in a product is a trivial topological fiber bundle. -/
+lemma is_trivial_topological_fiber_bundle_snd :
+  is_trivial_topological_fiber_bundle F (prod.snd : F × B → B) :=
+⟨homeomorph.prod_comm _ _, λ x, rfl⟩
 
 /-- The second projection in a product is a topological fiber bundle. -/
 lemma is_topological_fiber_bundle_snd : is_topological_fiber_bundle F (prod.snd : F × B → B) :=
-begin
-  let F : bundle_trivialization F (prod.snd : F × B → B) :=
-  { base_set      := univ,
-    open_base_set := is_open_univ,
-    source_eq     := rfl,
-    target_eq     := by simp,
-    proj_to_fun   := λp, by { simp, refl },
-     ..(homeomorph.prod_comm F B).to_local_homeomorph },
-  exact λx, ⟨F, by simp⟩
-end
+is_trivial_topological_fiber_bundle_snd.is_topological_fiber_bundle
+
+/-- Composition of a `bundle_trivialization` and a `homeomorph`. -/
+def bundle_trivialization.comp_homeomorph {Z' : Type*} [topological_space Z']
+  (e : bundle_trivialization F proj) (h : Z' ≃ₜ Z) :
+  bundle_trivialization F (proj ∘ h) :=
+{ to_local_homeomorph := h.to_local_homeomorph.trans e.to_local_homeomorph,
+  base_set := e.base_set,
+  open_base_set := e.open_base_set,
+  source_eq := by simp [e.source_eq, preimage_preimage],
+  target_eq := by simp [e.target_eq],
+  proj_to_fun := λ p hp,
+    have hp : h p ∈ e.source, by simpa using hp,
+    by simp [hp] }
+
+lemma is_topological_fiber_bundle.comp_homeomorph {Z' : Type*} [topological_space Z']
+  (e : is_topological_fiber_bundle F proj) (h : Z' ≃ₜ Z) :
+  is_topological_fiber_bundle F (proj ∘ h) :=
+λ x, let ⟨e, he⟩ := e x in
+⟨e.comp_homeomorph h, by simpa [bundle_trivialization.comp_homeomorph] using he⟩
+
+section induced
+
+open_locale classical
+
+variables {B' : Type*} [topological_space B']
+
+/-- Fiven a bundle trivialization of `proj : Z → B` and a continuous map `f : B' → B`,
+construct a bundle trivialization of `φ : {p : B' × Z | f p.1 = proj p.2} → B'`
+given by `φ x = (x : B' × Z).1`. -/
+noncomputable def bundle_trivialization.induced
```

Comment:
To me, this is the pullback of the local trivialization. We don't really use pullback in mathlib, but we use `comap` a lot. The problem I see with `induced` is that it makes me rather think of a restriction to an open set.